client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -0,0 +1,178 @@
+
+	// Check whether the wpcom.com subdomain warning is present.
+	const wordPressSubdomainWarning = allEligibilityWarnings?.find(
+		( { id } ) => id === 'wordpress_subdomain'
+	);
+
+	const eligibilityWarnings = allEligibilityWarnings?.filter(
+		( { id } ) => id !== 'wordpress_subdomain'
+	);
+
+	// Pick the wpcom subdomain.
+	const wpcomDomain = useSelector( ( state: AppState ) => getSiteDomain( state, siteId ) );
+
+	const isProcessing = ! siteId || isFetchingTransferStatus || ! wordPressSubdomainWarning;
+
+	function getWPComSubdomainWarningContent() {
+		// Get staging sudomain based on the wpcom subdomain.
+		const stagingDomain = wpcomDomain?.replace( /\b.wordpress.com/, '.wpcomstaging.com' );
+
+		return (
+			<>
+				<div className="confirm__image-container">
+					{ isProcessing && <LoadingEllipsis /> }
+					<img src={ yourNewStoreImage } alt="" />
+				</div>
+				<div className="confirm__instructions-container">
+					<div className="confirm__instructions-title confirm__instructions-wpcom-domain">
+						{ wpcomDomain }
+					</div>
+
+					<div className="confirm__instructions-title">{ stagingDomain }</div>
+
+					<p>
+						{ __(
+							'By installing this product your subdomain will change. You can change it later to a custom domain and we will pick up the tab for a year.'
+						) }
+					</p>
+
+					<p>
+						{ createInterpolateElement( __( '<a>Contact support</a> for help and questions.' ), {
+							a: <a href="#support-link" />,
+						} ) }
+					</p>
+
+					<NextButton
+						disabled={ isProcessing }
+						onClick={ () => {
+							if ( eligibilityHolds?.length || eligibilityWarnings?.length ) {
+								return goToStep( 'confirm', 'eligibility_substep' );
+							}
+
+							return goToStep( 'transfer' );
+						} }
+					>
+						{ __( 'Sounds good' ) }
+					</NextButton>
+				</div>
+			</>
+		);
+	}
+
+	function getContent() {
+		// wpcom subdomain warning.
+		if (
+			wordPressSubdomainWarning &&
+			( stepSectionName === 'wpcom_subdomain_substep' || typeof stepSectionName === 'undefined' )
+		) {
+			return getWPComSubdomainWarningContent();
+		}
+
+		return (
+			<>
+				<div className="confirm__image-container">
+					{ isProcessing && <LoadingEllipsis /> }
+					<img src={ yourNewStoreImage } alt="" />
+					<div>
+						<BackButton onClick={ () => goToStep( 'confirm', 'wpcom_subdomain_substep' ) } />
+					</div>
+				</div>
+				<div className="confirm__instructions-container">
+					{ !! eligibilityHolds?.length && (
+						<p>
+							<HoldList holds={ eligibilityHolds } context={ 'plugins' } isPlaceholder={ false } />
+						</p>
+					) }
+					{ !! eligibilityWarnings?.length && (
+						<p>
+							<WarningList warnings={ eligibilityWarnings } context={ 'plugins' } />
+						</p>
+					) }
+
+					<NextButton onClick={ () => goToStep( 'transfer' ) }>{ __( 'Confirm' ) }</NextButton>
+				</div>
+			</>
+		);
+	}
+
+	return (
+		<StepWrapper
```

Comment:
Should we pass in a `stepName` prop here? I'm seeing this warning and it looks like `StepWrapper` passes `stepName` to `NavigationLink`.
```
Warning: Failed prop type: The prop `stepName` is marked as required in `NavigationLink`, but its value is `undefined`.
```

Reply:
Adding stepName fixed this for me, if you add the step-components patch it gets passed straight into the confirm component as a prop from the singup framework.